example/Index.re

```diff
@@ -1,25 +1,25 @@
 open Express;
 
-/* The tests below relies upon the ability to store in the Request 
-   objects abritrary JSON properties. 
+/* The tests below relies upon the ability to store in the Request
+   objects abritrary JSON properties.
 
-   Each middleware will both check that previous middleware 
-   have been called by making properties exists in the Request object and 
-   upon success will themselves adds anothe property to the Request. 
+   Each middleware will both check that previous middleware
+   have been called by making properties exists in the Request object and
+   upon success will themselves adds anothe property to the Request.
 
 */
 
-/* [checkProperty req next property k] makes sure [property] is 
-   present in [req]. If success then [k()] is invoked, otherwise 
+/* [checkProperty req next property k] makes sure [property] is
+   present in [req]. If success then [k()] is invoked, otherwise
    [Next.route] is called with next */
 let checkProperty req next property k => {
-  let reqData = Request.asJsonObject req; 
+  let reqData = Request.asJsonObject req;
   switch (Js.Dict.get reqData property) {
     | None => next Next.route
     | Some x => {
       switch (Js.Json.decodeBoolean x) {
-        | Some b when (b == Js.true_) => k ()
-        | _ => next Next.route
+        | Some b when (b == Js.true_) => k ();
+        | _ => next Next.route;
```

Comment:
It's not a JSON object though. If you were to try to enumerate and deserialize the keys of the Request object as if it were JSON, it would fail.

With convenience functions this:
```reason
let reqData = Request.asJsonObject req;
switch (Js.Dict.get reqData property) {
| None => next Next.route
| Some x => {
  switch (Js.Json.decodeBoolean x) {
  | Some b when (b == Js.true_) => k ();
  | _ => next Next.route;
  }
}
```
Could be reduced to just
```reason
switch (req |> Request.getBoolean property) {
| Some b when b => k ();
| _ => next Next.route;
}
```

So it's both safer and more convenient, but of course it's more code and a slightly thicker layer of abstraction.